pkg/gather/tasks_processing.go

```diff
@@ -29,31 +29,42 @@ type GatheringFunctionResult struct {
 // HandleTasksConcurrently processes tasks concurrently and returns iterator like channel with the results
 // current implementation runs N goroutines where N is the number of tasks
 func HandleTasksConcurrently(ctx context.Context, tasks []Task) chan GatheringFunctionResult {
-	// TODO: consider using tasks pool with limited number of tasks run at the same time
-	// so that memory usage doesn't grow linearly with the number of added gatherers
-
 	resultsChan := make(chan GatheringFunctionResult)
 
 	// run all the tasks in the background and close the channel when they are finished
 	go func() {
 		var wg sync.WaitGroup
+		workerNum := 5 // TODO: Get this from config
```

Comment:
I think we want to find some better number by benchmarking for example. Cuz 5 threads should be pretty slow. Probably it should be runtime.NumCPU() multiplied by some number.

Reply:
I don't think it's necessary to make it configurable. WDYT?

Reply:
Probably not, but it should be at least moved to a constant.

Reply:
I choose `5` because it was just the first random number that came to mind :sweat_smile:

I think making it configurable would be cool if anyone wanted to configure it  (ie. the user) but I doubt anyone would actually do that, so it might be a waste of effort. 
Making it "smart" by looking at the number of CPU's and amount of RAM would be also cool but I have no idea what kind of thresholds would I give. (ie. what is the cpu/ram number after which its ok to add a few extra workers) Also making it smart scares me a bit. :smile:

Reply:
We don't need to look at the amount of RAM, it'd be too complicated, we just need to choose the number (which can be multiplied by `runtime.NumCPU()`) big enough so that the operator processes everything in reasonable time, but not too big so that it doesn't consume way too much memory on average.